feature_selection_in_R.Rmd

```diff
@@ -0,0 +1,175 @@
+# Feature selection in R
+
+Zehui Wu
+
+```{r}
+library(ggcorrplot)
+library(caret)
+library(randomForest)
+library(gam)
+```
+
+## 0. Introduction
+
+Feature selection is one of the most important tasks to boost performance of machine learning models. Some of the benefits of doing feature selections include:
+
+-   Better Accuracy: removing irrelevant features let the models make decisions only using important features. In my experience, classification models can usually get 5 to 10 percent improvement in accuracy scores after feature selection.
+
+-   Avoid Overfitting: the models will not put weights in irrelevant features.
+
+-   Improve running time: decreasing the dimension of the data makes the models run faster.
+
+In this tutorial, I will introduce some of the intuitive and popular methods in R for feature selection.
+
+## 1. Correlation Matrix
+
+Correlation matrix is a popular method for feature selection. By using correlation matrix, we can see the correlation for each pair of numerical variables. we not only can filter out variables with low correlation to the dependent variable, but also can remove redundant variables by identifying highly correlated independent variables.
+
+To illustrate how to use this method, I use the Winequality regression dataset from UCI.
+
+```{r}
+# load the data
+winequality <- read.csv("http://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-white.csv", sep = ";")
+head(winequality)
+```
+
+I firstly use the cor() function to calculate correlation matrix.
+
+```{r}
+cor_matrix <- data.frame(cor(winequality))
+cor_matrix
+```
+
+## 1.1 heat map for correlation matrix
```

Comment:
If you want to include subsection headings, use one more hashtag, and remember to remove the numbers for here and all below.
```suggestion
### heat map for correlation matrix

...

### function to find redundant variables

...
```

Reply:
change has made

Reply:
Sry for being so late. File name is changed!